### PR TITLE
Upgrade Mona Sans and Hubot Sans to 1.0.1 + Heading and Text component updates

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

> [!IMPORTANT]  
> This PR contains breaking changes. 

Updates the libraries font files to use latest 1.0.1 release of Mona Sans and Hubot Sans. 

Hubot Sans is no longer dark-shipped, and both fonts are now exposed in `Heading` and `Text` components via a `font` prop.

**Review the visual diffs for before and after comparisons.**

## List of notable changes:

- **updated** Mona Sans font file, and replaced Alliance2 with Hubot Sans 
- **added** new `font` prop that defaults to Mona Sans
- **added** missing `extralight` weight option
- **added** new stories for viewing Mona Sans and Hubot Sans

## What should reviewers focus on?

- Check that the rendering of the fonts is correct
- Does anything look incorrect in the visual diffs?

## Steps to test:

1. Use the playground to experiment with new `font` options
1. [Review the Heading stories for Mona Sans and verify rendering appears correct](https://primer-4e9e924e15-26139705.drafts.github.io/storybook/?path=/story/components-heading--mona-sans)
1. [Review the Heading stories for Hubot Sans and verify rendering appears correct](https://primer-4e9e924e15-26139705.drafts.github.io/storybook/?path=/story/components-heading--hubot-sans)
1. [Review the Text stories for Mona Sans and verify rendering appears correct](https://primer-4e9e924e15-26139705.drafts.github.io/storybook/?path=/story/components-text--mona-sans)
1. [Review the Text stories for Hubot Sans and verify rendering appears correct](https://primer-4e9e924e15-26139705.drafts.github.io/storybook/?path=/story/components-text--hubot-sans)

## Supporting resources (related issues, external links, etc):

- [github/mona-sans](https://github.com/github/mona-sans)
- [github/hubot-sans](https://github.com/github/hubot-sans)

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

Heading story for Mona Sans
heading story

Text story for Mona Sans:

image

### CODE REVIEW

**Comment by joseph-lozano** [comment]
I understand the change going from `heavy` to black, but why also change `normal` to `regular`?

Either way, I think that should be called out in the changeset.

**Comment by rezrah** [comment]
> I understand the change going from `heavy` to black, but why also change `normal` to `regular`?
> 
> Either way, I think that should be called out in the changeset.

Regular is the preset name recommended on the font pages. I've called that out in the changeset as it was missing, thanks for catching that.

**Review by joseph-lozano — APPROVED** [review]

File: packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Longer-Title-1-linux.png
Comment (auareyou) [inline packages/react/src/SubdomainNavBar/SubdomainNavBar.visual.spec.ts-snapshots/Visual-Comparison-SubdomainNavBar-SubdomainNavBar-Longer-Title-1-linux.png:1]: Is this overflow in the buttons what @joseph-lozano was working on?

**Review by auareyou — APPROVED** [review]
Looks all good! Approving to move forward. I added a little comment but I'm almost certain that isn't relevant to this PR. ✨

**Comment by josepmartins** [comment]
@nsolerieu do you happen to know if we need to update the Mona Sans + Hubot Sans versions in Figma?



Screenshot 2023-10-11 at 18 36 39

**Comment by rezrah** [comment]
Decision has been made to defer the changes for replacing regular with normal, and heavy with black to avoid potential overlap with Primer naming conventions. We'll review this again during the v8 migration.

**Pull request merged**